src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -221,47 +229,81 @@ static SGAAssemblyResult runSGAModulesInSerial(final String sgaPath,
-        if(null == fileNameToDedup){
-            return new SGAAssemblyResult(null, runtimeInfo);
+        {//merge
+            moduleArgs.clear();
+            moduleArgs.add("--min-overlap");     moduleArgs.add(String.valueOf(MIN_OVERLAP_IN_FILTER_OVERLAP_ASSEMBLE));
+
+            nameOfFileToFeedToNextStep = runSGAModule(sgaPath, "fm-merge", ".merged.fa", new File(workingDir, nameOfFileToFeedToNextStep), workingDir, indexer, indexerArgs, moduleArgs, runtimeInfo, enableSTDIOCapture);
+            if( null == nameOfFileToFeedToNextStep ){ return new SGAAssemblyResult(null, runtimeInfo); }
         }
 
-        final File fileToDeduplicate = new File(tempWorkingDir, fileNameToDedup);
-        final String fileNameToAssemble = runAndStopEarly("rmdup", fileToDeduplicate, sga, tempWorkingDir, indexer, indexerArgs, runtimeInfo, enableSTDIOCapture);
-        if(null == fileNameToAssemble){
-            return new SGAAssemblyResult(null, runtimeInfo);
+        {//deduplicate
+            moduleArgs.clear();
+            nameOfFileToFeedToNextStep = runSGAModule(sgaPath, "rmdup", ".rmdup.fa", new File(workingDir, nameOfFileToFeedToNextStep), workingDir, indexer, indexerArgs, moduleArgs, runtimeInfo, enableSTDIOCapture);
+            if( null == nameOfFileToFeedToNextStep ){ return new SGAAssemblyResult(null, runtimeInfo); }
         }
 
-        final File fileToAssemble   = new File(tempWorkingDir, fileNameToAssemble);
-        final String contigsFileName = runAndStopEarly("assemble", fileToAssemble, sga, tempWorkingDir, indexer, indexerArgs, runtimeInfo, enableSTDIOCapture);
-        if(null == contigsFileName){
-            return new SGAAssemblyResult(null, runtimeInfo);
+        {//overlap
+            moduleArgs.clear();
+            moduleArgs.add("--min-overlap");   moduleArgs.add(String.valueOf(MIN_OVERLAP_IN_FILTER_OVERLAP_ASSEMBLE));
+            moduleArgs.add("--error-rate");    moduleArgs.add(String.valueOf(OVERLAP_STEP_ERROR_RATE));
+
+            nameOfFileToFeedToNextStep = runSGAModule(sgaPath, "overlap", ".asqg.gz", new File(workingDir, nameOfFileToFeedToNextStep), workingDir, null, null, moduleArgs, runtimeInfo, enableSTDIOCapture);
+            if( null == nameOfFileToFeedToNextStep ){ return new SGAAssemblyResult(null, runtimeInfo); }
+        }
+
+        {//assembly
+            moduleArgs.clear();
+            moduleArgs.add("--cut-terminal");         moduleArgs.add(String.valueOf(CUT_OFF_TERMINAL_BRANCHES_IN_N_ROUNDS));
+            moduleArgs.add("--max-gap-divergence");   moduleArgs.add(String.valueOf(ASSEMBLE_STEP_MAX_GAP_DIVERGENCE));
+            moduleArgs.add("--min-overlap");          moduleArgs.add(String.valueOf(MIN_OVERLAP_IN_FILTER_OVERLAP_ASSEMBLE));
+            moduleArgs.add("--min-branch-length");    moduleArgs.add(String.valueOf(ASSEMBLE_STEP_MIN_BRANCH_TAIL_LENGTH));
+            moduleArgs.add("--out-prefix");           moduleArgs.add(nameOfFileToFeedToNextStep.replace(".asqg.gz", ""));
```

Comment:
I'd move the `.replace()` call to line 279 when you actually set the variable, for consistency.